[
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810066.289900",
        "client_msg_id": "fd58e273-413e-48de-ad48-a796b716477f",
        "text": "Hi, thanks for the quick reply. It looks like this:\n",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "w\/c",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Hi, thanks for the quick reply. It looks like this:\n"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810066.290000",
        "client_msg_id": "fd58e273-413e-48de-ad48-a796b716477f",
        "text": "```Traceback (most recent call last):\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 387, in _make_request\n    six.raise_from(e, None)\n  File \"&lt;string&gt;\", line 2, in raise_from\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 383, in _make_request\n    httplib_response = conn.getresponse()\n  File \"\/usr\/lib\/python3.6\/http\/client.py\", line 1356, in getresponse\n    response.begin()\n  File \"\/usr\/lib\/python3.6\/http\/client.py\", line 307, in begin\n    version, status, reason = self._read_status()\n  File \"\/usr\/lib\/python3.6\/http\/client.py\", line 268, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"\/usr\/lib\/python3.6\/socket.py\", line 586, in readinto\n    return self._sock.recv_into(b)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/contrib\/pyopenssl.py\", line 317, in recv_into\n    raise timeout('The read operation timed out')\nsocket.timeout: The read operation timed out\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 641, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/util\/retry.py\", line 368, in increment\n    raise six.reraise(type(error), error, _stacktrace)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/packages\/six.py\", line 686, in reraise\n    raise value\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 603, in urlopen\n    chunked=chunked)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 389, in _make_request\n    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 307, in _raise_timeout\n    raise ReadTimeoutError(self, url, \"Read timed out. (read timeout=%s)\" % timeout_value)\nurllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='<http:\/\/api.betfair.com|api.betfair.com>', port=443): Read timed out. (read timeout=16)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/betfairlightweight\/endpoints\/baseendpoint.py\", line 43, in request\n    timeout=(self.connect_timeout, self.read_timeout)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/api.py\", line 116, in post\n    return request('post', url, data=data, json=json, **kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/api.py\", line 60, in request\n    return session.request(method=method, url=url, **kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/adapters.py\", line 529, in send\n    raise ReadTimeout(e, request=request)\nrequests.exceptions.ReadTimeout: HTTPSConnectionPool(host='<http:\/\/api.betfair.com|api.betfair.com>', port=443): Read timed out. (read timeout=16)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"get_new_bets_minutely.py\", line 159, in &lt;module&gt;\n    main1()\n  File \"get_new_bets_minutely.py\", line 48, in main1\n    include_item_description = \"true\"\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/betfairlightweight\/endpoints\/betting.py\", line 288, in list_cleared_orders\n    (response, elapsed_time) = self.request(method, params, session)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/betfairlightweight\/endpoints\/baseendpoint.py\", line 48, in request\n    raise APIError(None, method, params, e)\nbetfairlightweight.exceptions.APIError: SportsAPING\/v1.0\/listClearedOrders \nParams: {'recordCount': 500, 'fromRecord': 1500, 'includeItemDescription': 'true', 'settledDateRange': {'to': None, 'from': None}, 'betStatus': 'SETTLED'} \nException: HTTPSConnectionPool(host='<http:\/\/api.betfair.com|api.betfair.com>', port=443): Read timed out. (read timeout=16)```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "w\/c-AYP",
                "elements": [
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Traceback (most recent call last):\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 387, in _make_request\n    six.raise_from(e, None)\n  File \"<string>\", line 2, in raise_from\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 383, in _make_request\n    httplib_response = conn.getresponse()\n  File \"\/usr\/lib\/python3.6\/http\/client.py\", line 1356, in getresponse\n    response.begin()\n  File \"\/usr\/lib\/python3.6\/http\/client.py\", line 307, in begin\n    version, status, reason = self._read_status()\n  File \"\/usr\/lib\/python3.6\/http\/client.py\", line 268, in _read_status\n    line = str(self.fp.readline(_MAXLINE + 1), \"iso-8859-1\")\n  File \"\/usr\/lib\/python3.6\/socket.py\", line 586, in readinto\n    return self._sock.recv_into(b)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/contrib\/pyopenssl.py\", line 317, in recv_into\n    raise timeout('The read operation timed out')\nsocket.timeout: The read operation timed out\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/adapters.py\", line 449, in send\n    timeout=timeout\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 641, in urlopen\n    _stacktrace=sys.exc_info()[2])\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/util\/retry.py\", line 368, in increment\n    raise six.reraise(type(error), error, _stacktrace)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/packages\/six.py\", line 686, in reraise\n    raise value\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 603, in urlopen\n    chunked=chunked)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 389, in _make_request\n    self._raise_timeout(err=e, url=url, timeout_value=read_timeout)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/urllib3\/connectionpool.py\", line 307, in _raise_timeout\n    raise ReadTimeoutError(self, url, \"Read timed out. (read timeout=%s)\" % timeout_value)\nurllib3.exceptions.ReadTimeoutError: HTTPSConnectionPool(host='api.betfair.com', port=443): Read timed out. (read timeout=16)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/betfairlightweight\/endpoints\/baseendpoint.py\", line 43, in request\n    timeout=(self.connect_timeout, self.read_timeout)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/api.py\", line 116, in post\n    return request('post', url, data=data, json=json, **kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/api.py\", line 60, in request\n    return session.request(method=method, url=url, **kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/sessions.py\", line 533, in request\n    resp = self.send(prep, **send_kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/sessions.py\", line 646, in send\n    r = adapter.send(request, **kwargs)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/requests\/adapters.py\", line 529, in send\n    raise ReadTimeout(e, request=request)\nrequests.exceptions.ReadTimeout: HTTPSConnectionPool(host='api.betfair.com', port=443): Read timed out. (read timeout=16)\n\nDuring handling of the above exception, another exception occurred:\n\nTraceback (most recent call last):\n  File \"get_new_bets_minutely.py\", line 159, in <module>\n    main1()\n  File \"get_new_bets_minutely.py\", line 48, in main1\n    include_item_description = \"true\"\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/betfairlightweight\/endpoints\/betting.py\", line 288, in list_cleared_orders\n    (response, elapsed_time) = self.request(method, params, session)\n  File \"\/home\/ubuntu\/.local\/lib\/python3.6\/site-packages\/betfairlightweight\/endpoints\/baseendpoint.py\", line 48, in request\n    raise APIError(None, method, params, e)\nbetfairlightweight.exceptions.APIError: SportsAPING\/v1.0\/listClearedOrders \nParams: {'recordCount': 500, 'fromRecord': 1500, 'includeItemDescription': 'true', 'settledDateRange': {'to': None, 'from': None}, 'betStatus': 'SETTLED'} \nException: HTTPSConnectionPool(host='api.betfair.com', port=443): Read timed out. (read timeout=16)"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810256.291000",
        "client_msg_id": "3c6eba19-6269-47e7-9441-efa73cc37c2d",
        "text": "It used to work fine, but then I found a few weeks back it wouldnt work for a few hours around 2pm UTC time. Now it never works.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "BQwh",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "It used to work fine, but then I found a few weeks back it wouldnt work for a few hours around 2pm UTC time. Now it never works."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1592810289.291400",
        "client_msg_id": "f9ed21d1-5970-4e72-868a-2f8fef021df6",
        "text": "What date range are you using?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "AI6",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "What date range are you using?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810305.292000",
        "client_msg_id": "56be0ab3-5f8a-416b-a8e7-8f2d9828dfdd",
        "text": "No date range. I just assumed it would return record 1500 - 2000",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "1Kq",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "No date range. I just assumed it would return record 1500 - 2000"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1592810305.292100",
        "client_msg_id": "24e2b0f4-062f-424c-87a9-b2997bfedb78",
        "text": "Ah right you're getting everything every time",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "+a4Y",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Ah right you're getting everything every time"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810335.292600",
        "client_msg_id": "4341bedc-c8a5-408c-880a-1decb063b5ca",
        "text": "Do you think the date range is the issue?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "gTPfy",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Do you think the date range is the issue?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1592810338.292700",
        "edited": {
            "user": "U4H19D1D2",
            "ts": "1592810391.000000"
        },
        "client_msg_id": "15b0795b-8891-47ae-9e92-bdf730e61d51",
        "text": "Can you try bumping the timeout\n\n```from betfairlightweight.endpoints.betting import Betting\n\nBetting.read_timeout = 32```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "bRizX",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Can you try bumping the timeout\n\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "from betfairlightweight.endpoints.betting import Betting\n\nBetting.read_timeout = 32"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1592810349.293100",
        "client_msg_id": "63023437-352e-4269-8be5-c06acac9447d",
        "text": "Yes, absolutely",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "B91",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Yes, absolutely"
                            }
                        ]
                    }
                ]
            }
        ],
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "U4H19D1D2"
                ],
                "count": 1
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810418.294400",
        "client_msg_id": "a4546dac-3171-4e7f-b64b-adf74b02ca0a",
        "text": "i tried doing `client.betting.read_timeout = 45` where client is an `APIClient` object, but that returned\n```Error: {'code': -32099, 'message': 'ANGX-0010', 'data': {'APINGException': {'requestUUID': 'ie1-ang07b-prd-05191058-003bfd', 'errorCode': 'TIMEOUT_ERROR', 'errorDetails': ''}, 'exceptionname': 'APINGException'}} Full Response: {'jsonrpc': '2.0', 'error': {'code': -32099, 'message': 'ANGX-0010', 'data': {'APINGException': {'requestUUID': 'ie1-ang07b-prd-05191058-003d', 'errorCode': 'TIMEOUT_ERROR', 'errorDetails': ''}, 'exceptionname': 'APINGException'}}, 'id': 1}```",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "dIZ",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "i tried doing "
                            },
                            {
                                "type": "text",
                                "text": "client.betting.read_timeout = 45",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " where client is an "
                            },
                            {
                                "type": "text",
                                "text": "APIClient",
                                "style": {
                                    "code": true
                                }
                            },
                            {
                                "type": "text",
                                "text": " object, but that returned\n"
                            }
                        ]
                    },
                    {
                        "type": "rich_text_preformatted",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Error: {'code': -32099, 'message': 'ANGX-0010', 'data': {'APINGException': {'requestUUID': 'ie1-ang07b-prd-05191058-003bfd', 'errorCode': 'TIMEOUT_ERROR', 'errorDetails': ''}, 'exceptionname': 'APINGException'}} Full Response: {'jsonrpc': '2.0', 'error': {'code': -32099, 'message': 'ANGX-0010', 'data': {'APINGException': {'requestUUID': 'ie1-ang07b-prd-05191058-003d', 'errorCode': 'TIMEOUT_ERROR', 'errorDetails': ''}, 'exceptionname': 'APINGException'}}, 'id': 1}"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1592810425.294700",
        "client_msg_id": "007cd6a0-5961-4d9f-adc0-a2be74f8b77a",
        "text": "The timeout will help, whether it's enough will depend on your level of betting activity",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "wkUw",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "The timeout will help, whether it's enough will depend on your level of betting activity"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810430.295000",
        "client_msg_id": "866c6d65-f9a7-4739-903a-53da7081bf3a",
        "text": "ok thanks <@UBS7QANF3> I'll try specifying a date range",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "739pJ",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "ok thanks "
                            },
                            {
                                "type": "user",
                                "user_id": "UBS7QANF3"
                            },
                            {
                                "type": "text",
                                "text": " I'll try specifying a date range"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1592810440.295300",
        "client_msg_id": "1fb80a1d-43aa-4e40-9eb2-33e73aee388a",
        "text": "But you're basically asking for 3 months worth of data every call",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "zcAcy",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "But you're basically asking for 3 months worth of data every call"
                            }
                        ]
                    }
                ]
            }
        ],
        "reactions": [
            {
                "name": "+1",
                "users": [
                    "UTP7UTQHK"
                ],
                "count": 1
            }
        ]
    },
    {
        "user": "UTP7UTQHK",
        "type": "message",
        "ts": "1592810793.295600",
        "client_msg_id": "d4cd0c16-d6d3-41a2-a737-cf0c543ddf3c",
        "text": "that worked - thanks heaps <@UBS7QANF3>",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g03753fb84ce",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/03753fb84ce45a184d8d1622be400be3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0000-72.png",
            "first_name": "JK",
            "real_name": "JK",
            "display_name": "JK",
            "team": "T4G9NBD2M",
            "name": "jhgward94",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "PyLkt",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "that worked - thanks heaps "
                            },
                            {
                                "type": "user",
                                "user_id": "UBS7QANF3"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UBS7QANF3",
        "type": "message",
        "ts": "1592811365.295800",
        "client_msg_id": "ebca0a36-f8ce-4e32-9238-8debef41c025",
        "text": "No worries",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gaaf844a4a90",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/eaaf844a4a905431d83430e563b077aa.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0011-72.png",
            "first_name": "",
            "real_name": "Maurice Berk",
            "display_name": "Mo",
            "team": "T4G9NBD2M",
            "name": "maurice",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "4\/N",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "No worries"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UFTBRB3F1",
        "type": "message",
        "ts": "1592837707.300300",
        "client_msg_id": "f7a0fedb-b541-4259-9a0c-ecaafc5428ab",
        "text": "Placing a single bet, I just got back an Error Code of 'ERROR_IN_ORDER', which is a fairly common one to see, but normally it is because another order has an 'INVALID_BET_SIZE' or something along those lines.\n\nIn this case however, that was the only error code (and it was just the lone bet)...looking at the timestamp it was sent just before the market was suspended (i.e. I received a market_book update @ 15:27:16.626763, with the next one coming @ 15:27:16.721394, showing it had been suspended (the place order instruction was sent just before the update).\n\nIs that error consistent with the bet being rejected due to a market being suspended (I'm assuming this is correct behaviour, but just wanted to check due to the non-descriptive error message)?",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g771e110be94",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/a771e110be945b86bede9d5c1bb551e3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0024-72.png",
            "first_name": "",
            "real_name": "Newbie99",
            "display_name": "",
            "team": "T4G9NBD2M",
            "name": "vjmehra",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Tht5s",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Placing a single bet, I just got back an Error Code of 'ERROR_IN_ORDER', which is a fairly common one to see, but normally it is because another order has an 'INVALID_BET_SIZE' or something along those lines.\n\nIn this case however, that was the only error code (and it was just the lone bet)...looking at the timestamp it was sent just before the market was suspended (i.e. I received a market_book update @ 15:27:16.626763, with the next one coming @ 15:27:16.721394, showing it had been suspended (the place order instruction was sent just before the update).\n\nIs that error consistent with the bet being rejected due to a market being suspended (I'm assuming this is correct behaviour, but just wanted to check due to the non-descriptive error message)?"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "subtype": "channel_join",
        "user": "U0160E9HS2G",
        "text": "<@U0160E9HS2G> has joined the channel",
        "type": "message",
        "ts": "1592838670.300500"
    },
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1592839840.301400",
        "client_msg_id": "D9319928-BE77-4E40-8967-5121B109F7EF",
        "text": "Yeah this can sometimes happen, gave up trying to get betfair to fix ",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Pf7Lv",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Yeah this can sometimes happen, gave up trying to get betfair to fix "
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UFTBRB3F1",
        "type": "message",
        "ts": "1592840037.301800",
        "client_msg_id": "ac34610c-580d-4fc7-ba62-7bfb42c8b71f",
        "text": "haha cool, good to know!",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g771e110be94",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/a771e110be945b86bede9d5c1bb551e3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0024-72.png",
            "first_name": "",
            "real_name": "Newbie99",
            "display_name": "",
            "team": "T4G9NBD2M",
            "name": "vjmehra",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "939T",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "haha cool, good to know!"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "subtype": "channel_join",
        "user": "U0167BJR4L9",
        "text": "<@U0167BJR4L9> has joined the channel",
        "type": "message",
        "ts": "1592849548.302000"
    },
    {
        "user": "U015LFSA4F8",
        "type": "message",
        "ts": "1592875540.303900",
        "client_msg_id": "c0d4aa0c-d091-4f00-bf09-8dee41fbbdca",
        "text": "<@UFTBRB3F1> whenever you get back a ERROR_IN_ORDER there should always be an accompanying reason.  This could be INVALID_BET_SIZE or MARKET_SUSPENDED, etc.  You need to check the reasn for the error.  Sometimes it can be a late scratching that has been removed from the market and you have updated your list of selections to see this change.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "gc56f45a421f",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/c56f45a421f6b7141d55bca979be523f.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0025-72.png",
            "first_name": "",
            "real_name": "Paul Smith",
            "display_name": "Paul Smith",
            "team": "T4G9NBD2M",
            "name": "paul.neil.smith",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "ucD",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "user",
                                "user_id": "UFTBRB3F1"
                            },
                            {
                                "type": "text",
                                "text": " whenever you get back a ERROR_IN_ORDER there should always be an accompanying reason.  This could be INVALID_BET_SIZE or MARKET_SUSPENDED, etc.  You need to check the reasn for the error.  Sometimes it can be a late scratching that has been removed from the market and you have updated your list of selections to see this change."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UFTBRB3F1",
        "type": "message",
        "ts": "1592892940.305100",
        "client_msg_id": "6A92AE68-50C7-439A-9F10-2AE79AB170AC",
        "text": "Yes that was my point, unlike normal there was no additional accompanying reason, but it sounds like an edge case from what Liam says.",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g771e110be94",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/a771e110be945b86bede9d5c1bb551e3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0024-72.png",
            "first_name": "",
            "real_name": "Newbie99",
            "display_name": "",
            "team": "T4G9NBD2M",
            "name": "vjmehra",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "ucBE",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Yes that was my point, unlike normal there was no additional accompanying reason, but it sounds like an edge case from what Liam says."
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "U4H19D1D2",
        "type": "message",
        "ts": "1592894060.305700",
        "client_msg_id": "01015653-964A-4002-8BD5-037DF21F6723",
        "text": "Yeah you need a bet to lapse right at the time you make a cancel and the market suspends ",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "137c5a3ef323",
            "image_72": "https:\/\/avatars.slack-edge.com\/2024-03-15\/6819395173841_137c5a3ef323f1944a1a_72.png",
            "first_name": "liam",
            "real_name": "liam",
            "display_name": "liam",
            "team": "T4G9NBD2M",
            "name": "liam",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "Wqp53",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "Yeah you need a bet to lapse right at the time you make a cancel and the market suspends "
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "user": "UFTBRB3F1",
        "type": "message",
        "ts": "1592895067.306100",
        "client_msg_id": "ae4a8bae-00bf-4a16-b090-ec5bb51a1ad5",
        "text": "yep, that sounds exactly what happened!",
        "team": "T4G9NBD2M",
        "user_team": "T4G9NBD2M",
        "source_team": "T4G9NBD2M",
        "user_profile": {
            "avatar_hash": "g771e110be94",
            "image_72": "https:\/\/secure.gravatar.com\/avatar\/a771e110be945b86bede9d5c1bb551e3.jpg?s=72&d=https%3A%2F%2Fa.slack-edge.com%2Fdf10d%2Fimg%2Favatars%2Fava_0024-72.png",
            "first_name": "",
            "real_name": "Newbie99",
            "display_name": "",
            "team": "T4G9NBD2M",
            "name": "vjmehra",
            "is_restricted": false,
            "is_ultra_restricted": false
        },
        "blocks": [
            {
                "type": "rich_text",
                "block_id": "bTsTs",
                "elements": [
                    {
                        "type": "rich_text_section",
                        "elements": [
                            {
                                "type": "text",
                                "text": "yep, that sounds exactly what happened!"
                            }
                        ]
                    }
                ]
            }
        ]
    },
    {
        "subtype": "channel_join",
        "user": "U0168CMAPUH",
        "text": "<@U0168CMAPUH> has joined the channel",
        "type": "message",
        "ts": "1592895449.306300"
    }
]